comp/metadata/inventoryhaagent/mock/mock.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package mock provides a mock for the inventoryhaagent component
+package mock
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+
+	inventoryhaagent "github.com/DataDog/datadog-agent/comp/metadata/inventoryhaagent/def"
+)
+
+// Module defines the fx options for the mock component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newMock))
+}
+
+type inventoryhaagentMock struct{}
+
+func newMock() inventoryhaagent.Component {
+	return &inventoryhaagentMock{}
+}
```

Comment:
The constructor should be public and must take a `testing.TB`/`testing.T` parameter even if we don't use it for now. That way we don't have to update the entire code base if we add a cleanup action to this mock

Reply:
I removed the mock for now, since not used.